aiida/engine/daemon/client.py

```diff
@@ -95,6 +102,14 @@ def daemon_name(self) -> str:
         """
         return self._DAEMON_NAME.format(name=self.profile.name)
 
+    @contextmanager
+    def lock(self):
+        if self.is_daemon_running:
```

Comment:
I think you are mixing the "old" check with the PID file with the "new" check with the lock file. However the logic sounds a bit complex to me, because the check on whether is locked is actually done in the `__init__`. Shouldn't this function raise if there is already a lock file? I.e. in the line with the `touch` below, and using `exist_ok=False`, catch that exception and re-raise the DaemonLock exception? Maybe I'm just missing how this lock manager would actually be used in practice (I know it's in the commit message but I'm still not sure of how you envision to use it in practice).

Reply:
Yeh I can do that as well 👍
The key point is to inhibit the instantiatuon of any further DaemonClient (from this or any other processes), since this is required for both `verdi daemon start-circus` and `verdi devel run-daemon`.
Naturally you could bypass this in some way, so it's not a "100% guarantee", but I think this is a 99% solution to provide a good level of insurance that no daemons are running while we clean the repo